wled00/FX.cpp

```diff
@@ -3385,58 +3391,57 @@ uint16_t mode_drip(void)
   struct virtualStrip {
     static void runStrip(uint16_t stripNr, Spark* drops) {
 
-      uint8_t numDrops = 1 + (SEGMENT.intensity >> 6); // 255>>6 = 3
+      uint8_t numDrops = map(SEGMENT.intensity, 0, 255, 1, 4);
 
-      float gravity = -0.0005 - (SEGMENT.speed/50000.0);
+      float gravity = -0.0005 - (SEGMENT.speed/25000.0); //increased max speed 50000->25000
       gravity *= SEGLEN-1;
-      int sourcedrop = 12;
+      // const int sourcedrop = 12;
 
       for (int j=0;j<numDrops;j++) {
-        if (drops[j].colIndex == 0) { //init
+        if (drops[j].colIndex == 0) { // drop state (0 init, 1 forming, 2 falling, 5 bouncing)
           drops[j].pos = SEGLEN-1;    // start at end
           drops[j].vel = 0;           // speed
-          drops[j].col = sourcedrop;  // brightness
-          drops[j].colIndex = 1;      // drop state (0 init, 1 forming, 2 falling, 5 bouncing)
+          drops[j].col = 0;           // brightness, start with 0
+          drops[j].velX = SEGMENT.color_from_palette(random8(), false, PALETTE_SOLID_WRAP, 0); // random color
```

Comment:
Assign velX to a uint8_t variable to avoid unnecessary float to byte conversions further down.

Reply:
done